docs/sdks/typescript/quickstart.md

```diff
@@ -142,365 +149,513 @@ function App() {
+  --textbox-color: #edfef4;
+  color-scheme: light dark;
+}
 
-1. `onConnect`: When we connect and receive our credentials, we'll save them to browser local storage, so that the next time we connect, we can re-authenticate as the same user.
-2. `initialStateSync`: When we're informed of the backlog of past messages, we'll sort them and update the `message` section of the page.
-3. `Message.onInsert`: When we receive a new message, we'll update the `message` section of the page.
-4. `User.onInsert`: When a new user joins, we'll update the `system` section of the page with an appropiate message.
-5. `User.onUpdate`: When a user is updated, we'll add a message with their new name, or declare their new online status to the `system` section of the page.
-6. `SetNameReducer.on`: If the server rejects our attempt to set our name, we'll update the `system` section of the page with an appropriate error message.
-7. `SendMessageReducer.on`: If the server rejects a message we send, we'll update the `system` section of the page with an appropriate error message.
+@media (prefers-color-scheme: dark) {
+  :root {
+    --theme-color: #4cf490;
+    --theme-color-contrast: #132219;
+    --textbox-color: #0f311d;
+  }
+}
 
-We will add callbacks for each of these items in the following sections. All of these callbacks will be registered inside the `App` function after the `useRef` declarations.
+/* ----- Page Setup ----- */
+html,
+body,
+#root {
+  height: 100%;
+  margin: 0;
+}
 
-### onConnect Callback
+body {
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
+    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
+    sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
 
-On connect SpacetimeDB will provide us with our client credentials.
+code {
+  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
+    monospace;
+}
 
-Each user has a set of credentials, which consists of two parts:
+/* ----- Buttons ----- */
+button {
+  padding: 0.5rem 0.75rem;
+  border: none;
+  border-radius: 0.375rem;
+  background-color: var(--theme-color);
+  color: var(--theme-color-contrast);
+  cursor: pointer;
+  font-weight: 600;
+  letter-spacing: 0.1px;
+  font-family: monospace;
+}
 
-- An `Identity`, a unique public identifier. We're using these to identify `User` rows.
-- A `Token`, a private key which SpacetimeDB uses to authenticate the client.
+/* ----- Inputs & Textareas ----- */
+input,
+textarea {
+  border: none;
+  border-radius: 0.375rem;
+  caret-color: var(--theme-color);
+  font-family: monospace;
+  font-weight: 600;
+  letter-spacing: 0.1px;
+  padding: 0.5rem 0.75rem;
+}
 
-These credentials are generated by SpacetimeDB each time a new client connects, and sent to the client so they can be saved, in order to re-connect with the same identity.
+input:focus,
+textarea:focus {
+  outline: none;
+  box-shadow: 0 0 0 2px var(--theme-color);
+}
+```
```

Comment:
We're spending a huge portion of this tutorial doing HTML and CSS related crap, which seems to me to provide no value to our goal of teaching the reader how to interact with SpacetimeDB.

Reply:
Correct, but I don't know how to avoid that. I didn't want to have to keep a starter project, a tutorial, and a complete project, all in sync. Each of these tutorials is a standalone document (you don't need anything else besides this document to complete it). I think that's a good practice.

Reply:
I think this is something we can improve in the future.  For now I don't love this, but I can live with it. As we get more time, more perspective and more automation, we can move to having a more succinct quickstart paired with more in-depth how-tos.